src/core/stream.c

```diff
@@ -713,7 +725,79 @@ QuicStreamParamGet(
 
         Status = QUIC_STATUS_SUCCESS;
         break;
+    case QUIC_PARAM_STREAM_BLOCKED_TIMINGS:
+        if (*BufferLength < sizeof(QUIC_FLOW_BLOCKED_TIMING) * QUIC_FLOW_BLOCK_REASON_COUNT) {
+            *BufferLength = sizeof(QUIC_FLOW_BLOCKED_TIMING) * QUIC_FLOW_BLOCK_REASON_COUNT;
+            Status = QUIC_STATUS_BUFFER_TOO_SMALL;
+            break;
+        }
+
+        if (Buffer == NULL) {
+            Status = QUIC_STATUS_INVALID_PARAMETER;
+            break;
+        }
 
+        uint64_t Now = CxPlatTimeUs64();
```

Comment:
I would really like if we could turn the code below into a loop somehow.

Reply:
It would be possible but I'd have to add a few more enums and make the tracking struct an array on stream and connection. Do you think it's worth it?

Reply:
I now have a strong feeling that this will make the code look way more unreadable and non-trivial after I just tried making the enum incremental and making QuicConnAddOutFlowBlockedReason adopt it.

So, I am gonna keep it this way unless you strongly insist.